dex/networks/btc/descriptors.go

```diff
@@ -0,0 +1,324 @@
+		}
+		var err error
+		d.Nested, err = parseDescriptor(arg, true)
+		if err != nil {
+			return nil, err
+		}
+		// If child SCRIPT was a KEY function, pull it up.
+		if d.Nested.KeyFmt != KeyUnknown {
+			d.KeyOrigin = d.Nested.KeyOrigin
+			d.Key = d.Nested.Key
+			d.KeyFmt = d.Nested.KeyFmt
+		}
+
+	default: // not KEY, like multi, tr, etc.
+	}
+
+	return d, nil
+}
+
+// ParseDescriptor parses a descriptor string. See
+// https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md
+// If the descriptor string does not match the expected pattern, an
+// ErrMalformedDescriptor error is returned. See the Descriptor type
+// documentation for information.
+func ParseDescriptor(desc string) (*Descriptor, error) {
+	return parseDescriptor(desc, false)
+}
+
+func parsePathPieces(pieces []string) (path []uint32, isRange bool, err error) {
+	for i, p := range pieces {
+		// In descriptor paths, "Anywhere a ' suffix is permitted to denote
+		// hardened derivation, the suffix h can be used instead."
+		if p == "*" || p == "*'" || p == "*h" {
+			if i != len(pieces)-1 {
+				return nil, false, errors.New("range indicator before end of path")
+			}
+			isRange = true
+			break // end of path, this is all the addresses
+		}
+		pN := strings.TrimRight(p, "'h") // strip any hardened character
+		pi, err := strconv.ParseUint(pN, 10, 32)
+		if err != nil {
+			return nil, false, err
+		}
+		if pN != p { // we stripped a ' or h => hardened
+			pi += uint64(hardened)
+		}
+		path = append(path, uint32(pi))
+	}
+	return
+}
+
+func parsePath(p string) (path []uint32, isRange bool, err error) {
+	if len(p) == 0 {
+		return
+	}
+	pp := strings.Split(p, "/")
+	return parsePathPieces(pp)
+}
+
+// ParsePath splits a common path string such as 84'/0'/0'/0 into it's
+// components, returning the steps in the derivation as integers. Hardened key
+// derivation steps, which are indicated by a ' or h, are offset by 2^31.
+// Certain paths in descriptors may end with /* (or /*' or /*h) to indicate all
+// child keys (or hardened child keys), in which case isRange will be true. As a
+// special case, a prefix of "m/" or just "/" is allowed, but a master key
+// fingerprint is not permitted and must be stripped first.
+func ParsePath(p string) (path []uint32, isRange bool, err error) {
+	p = strings.TrimPrefix(strings.TrimPrefix(p, "m"), "/")
+	return parsePath(p)
+}
+
+// DeepChild derives a new extended key from the provided root extended key and
+// derivation path. This is useful given a Descriptor.Key of type KeyExtended
+// that may be decoded into and extended key and path with ParseKeyExtended.
+// Given an address referencing the extended key via its fingerprint (also
+// returned by ParseKeyExtended) and derivation path, the private key for that
+// address may be derived given the child index of the address.
+func DeepChild(root *hdkeychain.ExtendedKey, path []uint32) (*hdkeychain.ExtendedKey, error) {
```

Comment:
I started that way but the dcrd hdkeychain package requires chaincfg and validates the network.  It's not the easiest to use in this case.

Reply:
Should we just use the `btcd` code in `keygen.go` then?

Reply:
We could, I think, since `keygen.go` doesn't use `NewKeyFromString` where the difference would mater (blake256 in the checksum instead of sha256).  We can look into that but I don't know if I wanna rock the boat there.
Will look at that.